File: heaps/minimum_operations_to_halve_array_sum.java
public int halveArray(int[] nums)
{
    double target = 0.0;
    PriorityQueue<Double> heap = new PriorityQueue<>(Comparator.reverseOrder());

    for (double num : nums)
    {
        target += num;
        heap.add(num);
    }

    target /= 2.0;

    int ans = 0.0;
    while (target > 0.0)
    {
        ans++;

        double x = heap.remove();
        target -= (x / 2.0);
        heap.add(x / 2.0);
    }

    return ans;
}

File: heaps/minimum_operations_to_halve_array_sum.py
from heapq import *


def halveArray(nums: list[int]) -> int:
    target = sum(nums) / 2
    heap = [-num for num in nums]
    heapify(heap)

    ans = 0
    while target > 0:
        ans += 1
        x = heappop(heap)
        target += x / 2
        heappush(heap, x / 2)
    
    return ans